### PR TITLE
Add OFFER coverage to portal-interop

### PR DESCRIPTION
**Pull request by KolbyML**
Time to add some coverage to offer tests.

### CODE REVIEW

File: simulators/portal-interop/src/main.rs [hunk 1]
```diff
@@ -67,49 +68,144 @@ dyn_async! {
 
         // Iterate over all possible pairings of clients and run the tests (including self-pairings)
         for (client_a, client_b) in clients.iter().cartesian_product(clients.iter()) {
-            // Test block header with proof
             test.run(
                 TwoClientTestSpec {
-                    name: format!("OFFER Block Header {} --> {}", client_a.name, client_b.name),
+                    name: format!("OFFER Header block 1 {} --> {}", client_a.name, client_b.name),
```
Comment (kdeme) [inline simulators/portal-interop/src/main.rs:73]: Just a little nitpick but perhaps you can consistently name all these it like this

```suggestion
                    name: format!("Offer Block Header: block number 1 {} --> {}", client_a.name, client_b.name),
```

**Review by kdeme — APPROVED** [review]

File: simulators/portal-interop/src/test-data/test_data_collection_of_forks_blocks.json
Comment (ogenev) [inline simulators/portal-interop/src/test-data/test_data_collection_of_forks_blocks.json:1]: My understanding is that we are trying to switch from `json` to `yaml` test files. How hard would be to convert this to yaml?

**Review by ogenev — CHANGES_REQUESTED** [review]
My understanding is that we are trying to switch from `json` to `yaml` test files as a general rule. How hard would be to convert this to `yaml` and add comments for each test vector?

**Comment by KolbyML** [comment]
@ogenev I changed it to yaml. Can you approve it now :shipit:

File: simulators/portal-interop/src/main.rs (hunk 1 above)
Comment (KolbyML) [reply simulators/portal-interop/src/main.rs]: Resolved

**Pull request merged**